cnijfilter2 no longer builds on top of master. The same is true for the version that's already there.

```
/nix/store/0gzd0049bywqhzi5anbydsxil7r1lfrj-binutils-2.38/bin/ld: cnijifnet2.o:(.bss+0x0): multiple definition of `GET_PROTOCOL'; cnijlgmon3.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:404: cnijlgmon3] Error 1
make[2]: Leaving directory '/build/source/lgmon3/src'
make[1]: *** [Makefile:416: all-recursive] Error 1
make[1]: Leaving directory '/build/source/lgmon3'
make: *** [Makefile:348: all] Error 2
error: build of '/nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv' on 'ssh://remote-build-user@sintra.home.arpa' failed: builder for '/nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv' failed with exit code 2;
       last 10 log lines:
       > mv -f .deps/cnijutil.Tpo .deps/cnijutil.Po
       > /nix/store/qm2lv1gpbyn0rsfai40cbvj3h4gz69yc-bash-5.1-p16/bin/bash ../libtool  --tag=CC   --mode=link gcc  -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2   -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > libtool: link: gcc -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2 -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o  -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > /nix/store/0gzd0049bywqhzi5anbydsxil7r1lfrj-binutils-2.38/bin/ld: cnijifnet2.o:(.bss+0x0): multiple definition of `GET_PROTOCOL'; cnijlgmon3.o:(.bss+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [Makefile:404: cnijlgmon3] Error 1
       > make[2]: Leaving directory '/build/source/lgmon3/src'
       > make[1]: *** [Makefile:416: all-recursive] Error 1
       > make[1]: Leaving directory '/build/source/lgmon3'
       > make: *** [Makefile:348: all] Error 2
       For full logs, run 'nix log /nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv'.
error: builder for '/nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv' failed with exit code 1;
       last 10 log lines:
       > mv -f .deps/cnijutil.Tpo .deps/cnijutil.Po
       > /nix/store/qm2lv1gpbyn0rsfai40cbvj3h4gz69yc-bash-5.1-p16/bin/bash ../libtool  --tag=CC   --mode=link gcc  -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2   -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > libtool: link: gcc -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2 -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o  -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > /nix/store/0gzd0049bywqhzi5anbydsxil7r1lfrj-binutils-2.38/bin/ld: cnijifnet2.o:(.bss+0x0): multiple definition of `GET_PROTOCOL'; cnijlgmon3.o:(.bss+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [Makefile:404: cnijlgmon3] Error 1
       > make[2]: Leaving directory '/build/source/lgmon3/src'
       > make[1]: *** [Makefile:416: all-recursive] Error 1
       > make[1]: Leaving directory '/build/source/lgmon3'
       > make: *** [Makefile:348: all] Error 2
       For full logs, run 'nix log /nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv'.
```

--- a/pkgs/misc/cups/drivers/cnijfilter2/default.nix
+++ b/pkgs/misc/cups/drivers/cnijfilter2/default.nix
@@ -4,11 +4,11 @@
 stdenv.mkDerivation {
   pname = "cnijfilter2";
 
-  version = "6.10";
+  version = "6.40";
 
   src = fetchzip {
-    url = "https://gdlp01.c-wss.com/gds/1/0100010921/01/cnijfilter2-source-6.10-1.tar.gz";
-    sha256 = "0w121issdjxdv5i9ksa5m23if6pz1r9ql8p894f1pqn16w0kw1ix";
+    url = "https://gdlp01.c-wss.com/gds/1/0100011381/01/cnijfilter2-source-6.40-1.tar.gz";
+    sha256 = "3RoG83jLOsdTEmvUkkxb7wa8oBrJA4v1mGtxTGwSowU=";
   };
 
   nativeBuildInputs = [ automake autoconf ];
@@ -25,7 +25,7 @@ stdenv.mkDerivation {
   # $out/lib/cups/filter/libcnbpcnclapicom2.so
   buildPhase = ''
     mkdir -p $out/lib
-    cp com/libs_bin64/* $out/lib
+    cp com/libs_bin_x86_64/* $out/lib
     mkdir -p $out/lib/cups/filter
     ln -s $out/lib/libcnbpcnclapicom2.so $out/lib/cups/filter
 
@@ -47,6 +47,12 @@ stdenv.mkDerivation {
 
     (
       cd cmdtocanonij2
+      ./autogen.sh --prefix=$out
+      make
+    )
+
+    (
+      cd cmdtocanonij3
       ./autogen.sh --prefix=$out
       make
     )
@@ -87,6 +93,11 @@ stdenv.mkDerivation {
 
     (
       cd cmdtocanonij2
+      make install
+    )
+
+    (
+      cd cmdtocanonij3
       make install
     )
 

--- a/pkgs/misc/cups/drivers/cnijfilter2/default.nix
+++ b/pkgs/misc/cups/drivers/cnijfilter2/default.nix
@@ -16,6 +16,10 @@ stdenv.mkDerivation {
     cups glib libxml2 libusb1 libtool
   ];
 
+  patches = [
+    ./patches/get_protocol.patch
+  ];
+
   # lgmon3's --enable-libdir flag is used soley for specifying in which
   # directory the cnnnet.ini cache file should reside.
   # NixOS uses /var/cache/cups, and given the name, it seems like a reasonable

--- a/pkgs/misc/cups/drivers/cnijfilter2/patches/get_protocol.patch
+++ b/pkgs/misc/cups/drivers/cnijfilter2/patches/get_protocol.patch
@@ -1,0 +1,14 @@
+# Resolves the compilation issue reported at https://github.com/NixOS/nixpkgs/pull/180681#issuecomment-1192304711
+# An identical issue was previously reported in Gentoo: https://bugs.gentoo.org/723186
+# This patch is taken from the solution of Alfredo Tupone (https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=24688d64544b43f2c14be54531ad8764419dde09)
+--- a/lgmon3/src/cnijlgmon3.c	2022-07-22 12:44:32.194641628 +0100
++++ b/lgmon3/src/cnijlgmon3.c	2022-07-22 12:43:53.954817724 +0100
+@@ -55,7 +55,7 @@
+ int (*GET_STATUS)(char *, int, int *, int * , char *);
+ int (*GET_STATUS2)(char *, int, char *, int *, int * , char *, char *);
+ int (*GET_STATUS2_MAINTENANCE)(char *, int, char *, int *, int * , char *, char *);
+-int (*GET_PROTOCOL)(char *, size_t);
++static int (*GET_PROTOCOL)(char *, size_t);
+ 
+ 
+  int main(int argc, char *argv[])

